newt/repo/version.go

```diff
@@ -126,6 +126,15 @@ func (r *Repo) VersionIsValid(ver newtutil.RepoVersion) bool {
 	}
 
 	cs, _ := r.downloader.CommitsFor(r.Path(), ver.Commit)
+
+	// For shallow copy commit might not be fetched yet
```

Comment:
this could be actually done regardless of whether repo is shallow or now
the same problem exists if user specifies a commit that is not preset on any remote branch, e.g. a sha for PR head